lib/environment/foreground/ajax.js

```diff
@@ -58,6 +58,14 @@ export async function ajax(options: AjaxOptions<*>) {
 		}
 	}
 
+	if (sameOrigin) {
+		if (method !== 'GET' && method !== 'HEAD') {
+			// Send modhash for same-origin non-GET requests
+			const hash = await loggedInUserHash();
```

Comment:
This block could have stayed in `buildRequestParams` if I made that function async. too. 

However, since `ajax.invalidate` works better with a synchronous function and doesn't care about the modhash header, it seems fine to move it into the main ajax function.